staging/src/k8s.io/apimachinery/pkg/api/validation/objectmeta.go

```diff
@@ -132,7 +132,7 @@ func ValidateNoNewFinalizers(newFinalizers []string, oldFinalizers []string, fld
 func ValidateImmutableField(newVal, oldVal interface{}, fldPath *field.Path) field.ErrorList {
 	allErrs := field.ErrorList{}
 	if !apiequality.Semantic.DeepEqual(oldVal, newVal) {
-		allErrs = append(allErrs, field.Invalid(fldPath, newVal, FieldImmutableErrorMsg))
+		allErrs = append(allErrs, field.ChangeForbidden(fldPath, newVal, FieldImmutableErrorMsg))
```

Comment:
@liggitt is this one of the places where we have to change to type of error produced in the validation functions.

Reply:
yes

Reply:
@liggitt Could you please also give your thoughts on [this](https://github.com/kubernetes/kubernetes/pull/108578#issuecomment-1061407326) comment (I updated it).

Reply:
this change looks about like what I would expect

Reply:
That said, there may be calls to `ValidateImmutableField` from update validation methods that are not even checking whether the data in the new object is valid or not because they are forbidding any change. We need to sweep calls to `ValidateImmutableField` and ensure callers also validate the data in the new object, in addition to blocking changes relative to the old object.

Reply:
> this sweep of calls to ValidateImmutableField is the next step and need to happen before we merge anything that would start returning ChangeForbidden errors.

this needs to happen in this PR before we make anything start returning ChangeForbidden errors

Reply:
/hold
To make sure this sweep happens before merge